dev/src/TestUserInput.ts

```diff
@@ -147,3 +159,25 @@ export class TestUserInput implements types.TestUserInput {
         return result;
     }
 }
+
+
+export async function runWithInputs<T>(callbackId: string, inputs: (string | RegExp | types.TestInput)[], registerOnActionStartHandler: types.registerOnActionStartHandlerType, callback: () => Promise<T>): Promise<T> {
+    const testUserInput = await TestUserInput.create();
+    testUserInput.setInputs(inputs);
+    const disposable = registerOnActionStartHandler((context) => {
+        if (context.callbackId === callbackId) {
```

Comment:
Just want to make sure I understand what's going on here.

So because `registerOnActionStartHandler` will let the handlers run in an arbitrary order, we're checking `callbackId` to ensure that this is the correct context to set the inputs for?

What will happen if it's not the same `callbackId`?  Will it just drop through and try to run the test anyway?

Reply:
No this doesn't have to do with multiple handlers, it has to do with multiple callbacks happening around the same time. Until this handler is disposed, every call to `callWithTelemetryAndErrorHandling` will also call this handler. If it's the expected callback id, we'll set the test user input and dispose the handler because we know we're done. If it's not, it'll just use whatever ui it already has and this does nothing.

For example, a single action can have multiple callbacks. The deploy action has at least 'azureFunctions.deploy' and 'postDeploy'. Plus there might be other random callbacks that happen like 'azureFunctions.validateFuncCoreToolsInstalled'